lib/tutorials/caching.md

```diff
@@ -0,0 +1,187 @@
+
+        var id = request.params.a + ':' + request.params.b;
+        sumCache.get({id: id, a: request.params.a, b: request.params.b}, function (err, result) {
+
+            reply(result);
+        })
+    }
+});
+```
+
+`cache` property tells hapi which [client](https://github.com/hapijs/catbox#client) to use.
+
+In addition to **partitions** there are **segments** that allows to isolate caches within one [client](https://github.com/hapijs/catbox#client). If you want to cache results from two different methods, you usually don't want mix up the results. In [mongoDB](http://www.mongodb.org/) `segment` represents collection and in [redis](http://redis.io/) it's additional prefix to `partition` prefix. Default value for `segment` is provided when [server.cache](http://hapijs.com/api#servercacheoptions) is called inside plugin - `'!pluginName'` or used via [server methods](http://hapijs.com/tutorials/server-methods) - `'#methodName'`. If you have use case for multiple policies sharing one segment there is [shared](http://hapijs.com/api#servercacheoptions) option.
+
+### Server methods
+But it can get better than that! In 95% cases you will use [server methods](http://hapijs.com/tutorials/server-methods) for caching purposes, because it reduces boilerplate to minimum. Lets do the same with server methods:
+
+```javascript
+server.method('sum', add, { cache: { cache: 'mongoCache', expiresIn: 30 * 1000 } });
+
+server.route({
+    path: '/add/{a}/{b}',
+    method: 'GET',
+    handler: function (request, reply) {
+
+        server.methods.sum(request.params.a, request.params.b, function (err, result) {
+
+            reply(result);
+        });
+    }
+});
+```
+
+Function [server.method()](http://hapijs.com/api#servermethodname-method-options) created new [policy](https://github.com/hapijs/catbox#policy) with `segment: '#sum'`. Also unique item `id` was automatically generated from parameters - by default it handles `string`,`number` and `boolean` parameters, for more complex parameters you have to provide own `generateKey` function, read more details in server methods [tutorial](http://hapijs.com/tutorials/server-methods).
+
+### What next?
+
+* Look into catbox policy [options](https://github.com/hapijs/catbox#policy). Especially to `staleIn`, `staleTimeout`, `generateTimeout`, which exploits full potential of catbox caching.
+* Check server methods [tutorial](http://hapijs.com/tutorials/server-methods) for more examples.
+
+## Client and Server caching
+
+[Catbox Policy](https://github.com/hapijs/catbox#policy) provides two more parameters `cached` and `report` which provides some details when item is cached.
+
+In this case we use `cached.stored` timestamp to set `last-modified` header.
+
+```javascript
+server.route({
+    path: '/add/{a}/{b}',
+    method: 'GET',
+    handler: function (request, reply) {
+
+        server.methods.sum(request.params.a, request.params.b, function (err, result, cached, report) {
```

Comment:
You could probably add these two parameters to the previous example so you didn't need to create a whole new one.